server/src/modules/auth/auth.service.ts

```diff
@@ -28,4 +35,17 @@ export class AuthService {
 
     return currentUser;
   }
+
+  refreshToken(refreshToken: string): ICurrentUser {
+    const currentUser: ICurrentUser = jwt.verify(
+      refreshToken,
+      process.env.REFRESH_JWT_SECRET,
+    ) as ICurrentUser;
+
+    if (currentUser) {
+      this.generateToken(currentUser);
```

Comment:
You should return refresh token, not `ICurrentUser`. This token should be sent to end user with access token.